lib/crack_pipe/action/exec.rb

```diff
@@ -80,13 +80,25 @@ def should_exec?(exec_if, action, context, kwargs)
         end
 
         def exec_with_args(e, action, context, kwargs)
+          action.context = context
+
           if e.is_a?(Symbol)
-            action.public_send(e, context, **kwargs)
+            if ctx_expected_as_arg?(action.method(e))
+              # DEPRECATED! :>
```

Comment:
What do you think about logging a warning for these deprecations?

Reply:
I would personally caution against doing this initially. We have a giant amount of code using the old way. If the migration to the new style turns out to be painless then I am okay with adding a warning.

Reply:
Being overwhelmed with warnings is for sure a risk/nuisance. The worry I have is if the deprecations aren't loud, we may delay making the changes and end up having both approaches for an extended/indefinite time.